dev/ledger.md

```diff
@@ -327,9 +307,113 @@ There exist two special addresses: $I_{pool}$, the address of the _incentive poo
 and $I_f$, the address of the _fee sink_.  For both of these accounts,
 $p_I = 2$.
 
+## Applications
+
+Each account can create applications, each named by a globally-unique integer
+(the _application ID_). Applications are associated with a set of _application
+parameters_, which can be encoded as a msgpack struct:
+
+- A mutable Stateful TEAL "Approval" program (`ApprovalProgram`), whose result
+  determines whether or not an `ApplicationCall` transaction referring to this
+  application ID is to be allowed. This program executes for all
+  `ApplicationCall` transactions referring to this application ID except for
+  those whose `OnCompletion == ClearState`, in which case the
+  `ClearStateProgram` is executed instead. This program may modify local or
+  global state associated with this application. This field is encoded with
+  msgpack field `approv`.
+
+  This field must not exceed 1024 bytes. The cost of the program as determined
+  by the Stateful TEAL `Check` function must not exceed 700.
+
+- A mutable Stateful TEAL "Clear State" program (`ClearStateProgram`), executed
+  when an opted-in user forcibly removes the local application state associated
+  with this application from their account data. This happens when an
+  `ApplicationCall` transaction referring to this application ID is executed
+  with `OnCompletion == ClearState`. This program, when executed, is not
+  permitted to cause the transaction to fail. This program may modify local or
+  global state associated with this application. This field is encoded with
+  msgpack field `clearp`.
+
+  This field must not exceed 1024 bytes. The cost of the program as determined
+  by the Stateful TEAL `Check` function must not exceed 700.
+
+- An immutable "global state schema" (`GlobalStateSchema`), which sets a limit
+  on the size of the global [TEAL Key/Value Store][TEAL Key/Value Stores] that
```

Comment:
I used the square bracket notation here to reference the section of the document called 'TEAL Key/Value Stores'. Is this what you mean? Or are you commenting on the term "TEAL Key/Value Store" specifically?

Reply:
Do you mean it to be a link with text "TEAL Key/Value Stores" that points to the section "TEAL Key/Value Store"? This is not how it displays in any of my browsers, Perhaps one of the parenthesis needs to be round rather than square?

Reply:
or maybe just make both of them "stores" (currently the second is "store")?

Reply:
(Ping @fabrice102 -- Discussed in the meeting today and it seems like we don't want to change this for now, but please let me know if I'm wrong!)